ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -1193,17 +1193,26 @@ generateGhcLibCabal ghcFlavor customCppOpts = do
         [ "source-repository head"
         , "    type: git"
         , "    location: git@github.com:digital-asset/ghc-lib.git"
-        , ""
-        , "library"
+        ] ++
+        [ "flag threaded-rts"
+        , "  default: True"
+        , "  manual: True"
+        , "  description: Pass -DTHREADED_RTS to the C toolchain"
+        ] ++
+        [ "library"
         , "    default-language:   Haskell2010"
         , "    exposed: False"
         , "    include-dirs:"
+        ] ++ indent2 (ghcLibIncludeDirs ghcFlavor) ++
+        [ "    if flag(threaded-rts)"
+        , "        ghc-options: -fobject-code -package=ghc-boot-th -optc-DTHREADED_RTS"
+        , "        cc-options: -DTHREADED_RTS"
+        , "        cpp-options: -DTHREADED_RTS " <> generateCppOpts ghcFlavor customCppOpts
+        , "    else"
+        , "        ghc-options: -fobject-code -package=ghc-boot-th"
+        , "        cpp-options: " <> generateCppOpts ghcFlavor customCppOpts
```

Comment:
`generateCppOpts` (defined in the same file) includes an entry for `-DTHREADED_RTS`:

https://github.com/digital-asset/ghc-lib/blob/2f65e35a6918c4b3e78373fe12b526175eb4338e/ghc-lib-gen/src/Ghclibgen.hs#L1228-L1232

so it would be included even if `threaded-rts` is disabled manually. It should be enough to remove the line in `generateCppOpts`.

Reply:
https://github.com/digital-asset/ghc-lib/pull/422/files#diff-472eaebd24c7e540a4d552d3b13cab6de8eb8d327990ae09778d91f30ae430f1L1232

Reply:
oh, somehow i missed that line, i'm sorry about that. No comments otherwise, thanks for this fix!